src/Rules/Arrays/ArrayDestructuringRule.php

```diff
@@ -66,7 +66,8 @@ private function getErrors(Scope $scope, Expr $var, Expr $expr): array
 			'',
 			static function (Type $varType): bool {
 				return $varType->isArray()->yes();
-			}
+			},
+			false
```

Comment:
Hi, can you please specify what was your framework about whether the shortcircuiting should take place or not? Thanks!

Reply:
It depends whether the caller of the helper method is interested in the final expression value or the possible value of the "prefix" expression.

Eg. difference between
$val = EXPR -> $val is of type VAL(EXPR), it does not matter whether the EXPR short-circuits or not
EXPR->methodCall() -> we're interested in the right-most value of the expression since if it short-circuits, the method would not be called.

Does it make sense this way?

Reply:
I understand, thank you, but I wonder if there's a more intuitive, or even automatic way to do this. As a first step, I'd like to ask you for adding tests for all the rules where you modified the RuleLevelHelper calls. Ideally the test should fail if I pass in the other boolean value.

Then we might be able to figure out an ideal API to do this.

We don't need to care about backward compatibility here - right now we're in 1.0-dev so BC breaks are allowed.

Reply:
I've added test cases for all rules that did not have null-safe operator test yet and rewritten the logic so that the short-circuiting happens in the rule itself and not in the RuleLevelHelper since it's the rule's responsibility to determine what it needs to check.